src/modules/polynomial/polynomial.cpp

```diff
@@ -0,0 +1,122 @@
+#include "polynomial.hpp"
+#include <iostream>
+
+Polynomial::Polynomial(): m_(0) {}
+
+Polynomial::Polynomial(std::vector<std::pair<Rational, int>> p) {
+    m_ = p[p.size() - 1].second;
+    c_.resize(m_);
+    for (std::pair cd : p)
+        c_[cd.second] = cd.first;
+}
+
+// P-3
+Polynomial Polynomial::MUL_PQ_P(const Rational& rational) const {
+    Polynomial result = *this;
+    // умножаем каждый коэффициент полинома
+    // на заданное число
+    for (Rational coeff : result.c_)
+        coeff = coeff.MUL_QQ_Q(rational);
```

Comment:
```suggestion
    for (auto &coeff : result.c_)
        coeff = coeff.MUL_QQ_Q(rational);
```
без ссылки на объект он не изменится
а еще хорошо бы после операции сделать нормализацию